runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -7969,7 +7969,7 @@ genInitObjectHeader(TR::Node * node, TR::Instruction *& iCursor, TR_OpaqueClassB
             }
          if (canUseIIHF)
             {
-            iCursor = generateRILInstruction(cg, TR::InstOpCode::IIHF, node, enumReg, (intptr_t) classAddress | (intptrj_t)orFlag, iCursor);
+            iCursor = generateRILInstruction(cg, TR::InstOpCode::IIHF, node, enumReg, static_cast<uint32_t>(reinterpret_cast<uintptr_t>(classAddress)) | static_cast<uint32_t>(orFlag), iCursor);
```

Comment:
Is there a typo here? Looks like we are mixing `uintptr_t` and `uint32_t` when doing the OR.

Reply:
Not a typo, although  see now the `static_cast` on the `orFlag` is unnecessary since it's declared as a `uint32_t`.

You can't use safe casting to cast `classAddress` to a `uint32_t`. It will complain:
`error: cast from 'TR_OpaqueClassBlock*' to 'uint32_t' loses precision`

The `OR` promotes the `orFlag` to a 64bit width so I needed to cast the result back down to `uint32_t` so it matches the overload.

Reply:
Thanks, was just making sure this was well thought out and it is! Yes the cast on `orFlag` is unnecessary.